### ISSUE TITLE
SMTP auth fails when username exceeds 31 characters

### ISSUE THREAD

**Issue by sysfu** [OpenSMTPD/OpenSMTPD#668]
I'm running into an issue on an OpenSMTPD mail server where the mail client cannot successfully authenticate via SMTP auth on port 587 when the username is longer than 31 characters.
 
Happens with both Mailbird and Thunderbird email clients.
 
Errors like the one below show up in /var/log/maillog when the username is over 31 chars and the client auth fails when attempting to send an email.
 
```
Mar 12 21:51:17 mx smtpd[26118]: smtp-in: Failed command on session 78fa93cb23366cfe: "AUTH PLAIN (...)" => 501 5.5.2 Syntax error: Syntax error
```
 
environment:
OpenBSD 5.8 release
LibreSSL 2.2.6
OpenSMTPD 5.7.3
Mailbird 2.2.5.0
Thunderbird 38.6.0
 
```
/etc/mail/smtpd.conf
 
pki domain.tld certificate "/etc/ssl/domain_tld.crt"
pki domain.tld key "/etc/ssl/private/domain_tld.key"
 
table creds             "/etc/mail/creds"
table trusted-relays    "/etc/mail/trusted-relays"
table enforce-tls       "/etc/mail/enforce-tls"
table vdoms             "/etc/mail/vdoms"
table vusers            "/etc/mail/vusers"
 
bounce-warn 1d
 
limit mta inet4
listen on lo inet4
listen on lo0 port 10028 tag DKIM_OUT # outgoing mail
listen on egress inet4 tls pki domain.tld
listen on egress inet4 port 587 tls-require pki domain.tld auth <creds> mask-source
 
# tagged mail returned from dkimproxy_out relays out
accept tagged DKIM_OUT for domain <enforce-tls> relay tls # require STARTTLS for domains known to support it, otherwise fail hard
accept tagged DKIM_OUT for any relay
 
# incoming mail for local users delivered via maildir
accept from any for domain <vdoms> virtual <vusers> deliver to maildir "/var/maildir/%{dest.domain:lowercase}/%{dest.user:lowercase|strip}/mail/"
 
# relay other email to the Internet at large
accept from source <trusted-relays> for any relay via smtp://127.0.0.1:10027
accept for any relay via smtp://127.0.0.1:10027
```

**Comment by poolpOrg** [comment]
Can you confirm this still happens with the code in OpenBSD 5.9 / git ?

I have spent time getting rid of the 31 chars limit and it's not supposed to be a problem anymore.

**Comment by sysfu** [comment]
Just to confirm, this issue should be resloved with the version of OpenSMTPD that shipped with the OpenBSD 5.9 release?

**Comment by poolpOrg** [comment]
This issue should be resolved with OpenSMTPD shipped in OpenBSD 5.9, yes

**Comment by sysfu** [comment]
Tested and confirmed issue is resolved, thank you.
